File: backend/app/schemas/routing.py
from pydantic import BaseModel


class CoordinateResponse(BaseModel):
    lat: float
    lon: float


class WalkingRouteResponse(BaseModel):
    origin: CoordinateResponse
    destination: CoordinateResponse
    distance_meters: float | None = None
    duration_seconds: float | None = None
    geometry: list[CoordinateResponse]
    provider: str

File: frontend/src/types/routing.ts
export interface RouteCoordinate {
  lat: number;
  lon: number;
}

export interface WalkingRouteResponse {
  origin: RouteCoordinate;
  destination: RouteCoordinate;
  distance_meters?: number | null;
  duration_seconds?: number | null;
  geometry: RouteCoordinate[];
  provider: string;
}
